src/components/storage/CockpitStorageIntegration.jsx

```diff
@@ -430,6 +432,66 @@ const CheckStorageDialog = ({
                 });
     }, [dispatch, checkStep, cockpitPassphrases, newMountPoints, devices, onCritFail, setError]);
 
+    useEffect(() => {
+        if (checkStep !== "mdarray") {
+            return;
+        }
+
+        const mdArrays = Object.keys(devices).filter(device => devices[device].type.v === "mdarray");
+
+        // In blivet we recognize two "types" of MD array:
+        // * The array is directly on top of disks: in this case we consider the array to be a disk
+        // (similar to a hardware RAID) and create the partition table on the array
+        // * The array is on top of partitions: from our pov this is a device and we allow only a single
+        // filesystem (or other format like lvmpv) on top of it and if it has partitions they are ignored
+        //
+        // For the first scenario, we need to re-set 'SelectedDisks' in backend,
+        // for the new mdarrays to be handled as such.
+        const selectedMDarrays = selectedDisks.map(disk => {
+            const mdArray = devices[disk].children.v.filter(child => mdArrays.includes(child));
+            return mdArray.length > 0 ? mdArray[0] : disk;
+        });
+
+        // Check if we have mdarrays that are not fitting in the above two scenarios
+        // and show an error message
+        const mdArraysNotSupported = mdArrays.filter(device => {
+            if (
+                devices[device].parents.v.every(parent => devices[parent].type.v === "disk") &&
+                devices[device].formatData.type.v === "disklabel"
+            ) {
+                return false;
+            }
+            if (
+                devices[device].parents.v.every(parent => devices[parent].type.v === "partition") &&
+                devices[device].formatData.type.v !== "disklabel"
+            ) {
+                return false;
+            }
+            return true;
+        });
+        // TODO: Consider moving this logic to the backend
+        if (mdArraysNotSupported.length > 0) {
+            setError({
+                message: cockpit.format(
+                    _(
+                        "Invalid RAID configuration detected.\n" +
+                        "If your RAID array is created directly on top of disks, a partition table must be created on the array.\n" +
```

Comment:
@vojtechtrefny we discussed this last week, I think we should move this logic elsewere, (blivet and fedora documentation)
But for resolving fast the bug, can you please review this error message?

Reply:
Strictly speaking, the first part is not true -- we support disks with filesystem without partition table, it's just not something generally recommend (and doesn't make much sense during installation anyway). But I am not really sure what will Anaconda do with disks like that (I think default behaviour is to create a new partition table on all selected disks) so I guess it makes sense to mark it as unsupported.